### PR TITLE
Split afvx to two commands - one for variables and one for function arguments #332

### PR DESCRIPTION
**Pull request by no1rr**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd_analysis.c [hunk 1]
```diff
@@ -851,7 +851,11 @@ static void var_accesses_list(RzAnalysisFunction *fcn, RzAnalysisVar *var, PJ *p
 	}
 }
 
-static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name) {
+#define IS_VAR 0
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: It is better to use enum for that.

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -9599,13 +9605,23 @@ RZ_IPI RzCmdStatus rz_analysis_function_vars_xrefs_handler(RzCore *core, int arg
 	} else {
 		rz_cons_printf("afvR\n");
 	}
-	list_vars(core, fcn, pj, 'R', argv[1]);
+	if (use_afvxa) {
+		list_vars(core, fcn, pj, 'R', argv[1], IS_ARG);
+	}		 
+    if (use_afvxv) {
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: Please use `sys/clang-format.py` to fix indentation.

File: librz/core/cmd_analysis.c [hunk 3]
```diff
@@ -9585,7 +9591,7 @@ RZ_IPI RzCmdStatus rz_analysis_function_vars_type_handler(RzCore *core, int argc
 	return RZ_CMD_STATUS_OK;
 }
 
-RZ_IPI RzCmdStatus rz_analysis_function_vars_xrefs_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+RZ_IPI RzCmdStatus RZ_ANALYSIS_FUNCTUIN_VARS_XREFS_HANDLER(RzCore *core, int argc, const char **argv, RzOutputMode mode, bool use_afvxa, bool use_afvxv) {
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: 1. A typo
2. Why CAPS?

**Review by XVilka — CHANGES_REQUESTED** [review]
I would avoid adding more parameters for this function as a solution.

File: librz/core/cmd_analysis.c (hunk 3 above)
Comment (no1rr) [reply librz/core/cmd_analysis.c]: Sorry, I don't know what CAPS means. My idea is to rename the original `rz_analysis_function_vars_xrefs_handler` to `RZ_ANALYSIS_FUNCTION_VARS_XREFS_HANDLER` and add three functions for `afvx`, `afvxa` and `afvxv` to call `RZ_ANALYSIS_` `FUNCTION_VARS_XREFS_HANDLER`

Comment (wargio) [reply librz/core/cmd_analysis.c]: CAPS means `all uppercase`. why you changed the name of the function to all uppercase letters? normally uppercased names are assigned to macros with constants

Comment (no1rr) [reply librz/core/cmd_analysis.c]: Thanks, I will rename the function in another way.

File: librz/core/cmd_analysis.c [hunk 4]
```diff
@@ -851,7 +851,11 @@ static void var_accesses_list(RzAnalysisFunction *fcn, RzAnalysisVar *var, PJ *p
 	}
 }
 
-static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name) {
+enum { IS_VAR = 0,
+	IS_ARG = 1,
+	IS_ARG_AND_VAR = 2 };
+
+static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name, int is_arg) {
```
Comment (wargio) [inline librz/core/cmd_analysis.c]:
```suggestion
typedef enum {
	IS_VAR = 0,
	IS_ARG,
	IS_ARG_AND_VAR
} RzVarListType;

static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name, RzVarListType is_arg) {
```

Comment (wargio) [reply librz/core/cmd_analysis.c]: i would also rename `is_arg` to something else.

File: librz/core/cmd_analysis.c [hunk 5]
```diff
@@ -9585,7 +9591,7 @@ RZ_IPI RzCmdStatus rz_analysis_function_vars_type_handler(RzCore *core, int argc
 	return RZ_CMD_STATUS_OK;
 }
 
-RZ_IPI RzCmdStatus rz_analysis_function_vars_xrefs_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+RZ_IPI RzCmdStatus rz_analysis_function_args_and_vars_xrefs_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode, bool use_afvxa, bool use_afvxv) {
```
Comment (wargio) [inline librz/core/cmd_analysis.c]: please rename `use_afvxa` to `use_args` and `use_afvxv` to `use_vars`

**Review by wargio — CHANGES_REQUESTED** [review]
be sure to run `./sys/clang-format.py` before committing.

File: librz/core/cmd_analysis.c [hunk 6]
```diff
@@ -851,11 +851,13 @@ static void var_accesses_list(RzAnalysisFunction *fcn, RzAnalysisVar *var, PJ *p
...
-	IS_ARG = 1,
-	IS_ARG_AND_VAR = 2 };
+typedef enum {
+	IS_VAR = 0,
+	IS_ARG,
+	IS_ARG_AND_VAR
+} RzVarListType;
 
-static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name, int is_arg) {
+static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, const char *name, RzVarListType varlisttype) {
```
Comment (wargio) [inline librz/core/cmd_analysis.c]: how about `varlisttype` > `vlt` ?

**Review by wargio — APPROVED** [review]
besides that detail, LGTM

**Review by XVilka — APPROVED** [review]
Lets wait for @ret2libc approval first since it's a command change, apart from that LGTM.

**Comment by XVilka** [comment]
@no1rr you forgot to regenerate C files - linter says they are out of sync.

**Comment by XVilka** [comment]
1. Please rebase on top of the latest `dev`.
2. Please build the Rizin first - it will regenerate *.c/*.h files
3. Commit/squash again.

**Comment by XVilka** [comment]
Also it's a good idea to use a different branch name usually.

**Comment by XVilka** [comment]
You still need to regenerate the C files: https://github.com/rizinorg/rizin/pull/970/checks?check_run_id=2287206397#step:5:1

**Comment by no1rr** [comment]
I built the Rizin and commited again, but linter still says C files are out of sync.

**Comment by XVilka** [comment]
I created new pull request here: https://github.com/rizinorg/rizin/pull/975

**Pull request closed without merging**